packages/block-editor/src/components/inserter/style.scss

```diff
@@ -433,7 +433,7 @@ $block-inserter-tabs-height: 44px;
 .block-editor-block-patterns-explorer {
 	&__sidebar {
 		position: absolute;
-		top: $header-height + $grid-unit-15;
+		top: $header-height + $grid-unit-10;
```

Comment:
Required in the Pattern Inserter modal to make the sidebar exactly adjacent to the modal header.

![patterns-sidebar](https://github.com/WordPress/gutenberg/assets/54422211/ac9a3ff0-3373-47dd-8b6d-524c5e49dbaf)